lib/resource.js

```diff
@@ -1,170 +1,104 @@
+        params.callback  = callback;
 
-    if (params && params.product) {
-      if (!parseInt(params.product, 10)) {
-        return params.product;
-      }
-    }
-
-    if (params && this.path.match(/videos/)) {
-      return params;
-    }
-
-    if (params && this.path.match(/customers/)) {
-      return params;
-    }
-
-    return `${this._api.protocol}${this._api.host}/${this.path}`;
-  }
-
-  getMethod(params) {
-    let _this = this;
-
-    if (_this.isObject(params) && params.method) {
-      return params.method;
-    } else {
-      return params;
-    }
-  }
-
-  needsToken() {
-    return new Date() > this._api.token_expiration;
+        this.createRequestParams(params);
+      };
+    });
   }
 
   getParams(client_method, url, options, scope, type) {
-    let _this  = this;
     let params = {};
 
-    params.timeout = _this._api.timeout;
-    params.qs      = options || null;
+    params.timeout = this._api.timeout;
     params.method  = client_method;
+    params.url     = url;
 
-    if (Object.prototype.toString.call(url) === '[object Object]') {
-      params.url = url[Object.keys(url)];
-    } else {
-      params.url = url;
-    }
-
-    if (params.url.match(/videos|customers/) && !params.url.match(/watching|watchlist/)) {
-      params.qs = '';
+    if (isObject(options)) {
+      params.qs = options || null;
     }
 
-    if (client_method === 'all') {
-      params.qs = '';
-    }
-
-    if (client_method === 'all' && options && parseInt(options.product, 10)) {
-      params.qs = options;
+    if (this._api.auth) {
+      params.headers = {
+        'Authorization': this._api.auth
+      };
     }
 
-    if (_this._api.auth) {
+    if (this._api.internal) {
       params.headers = {
-        'Authorization': _this._api.auth
-      };
+        'Authorization': 'Bearer ' + this._api.token
+      }
     }
 
     return params;
   }
 
-  makeRequest(args) {
-    let params = this.getParams(args.client_method, args.url, args.options, args.scope, args.read_only);
+  createRequestParams(args) {
+    const params = this.getParams(args.client_method, args.url, args.options, args.scope, args.read_only);
 
-    if (this.isCallbackFunction(args.options)) {
+    if (isFunction(args.options)) {
       args.callback = args.options;
     }
 
-    if (typeof this._api.token_host !== undefined) {
-      params.headers = {
-        'Authorization': 'Bearer ' + this._api.token
-      }
-
-      this.ajaxRequest(args, params);
-    } else {
-      this.ajaxRequest(args, params);
-    }
+    this.ajaxRequest(args, params);
   }
 
   ajaxRequest(args, params) {
-    let _this = this;
-
     request[args.http_method](params.url)
       .withCredentials()
       .set(params.headers || {})
       .set('Content-Type', 'application/json')
       .query(params.qs)
-      .end(function(err, response) {
+      .end((err, response) => {
         if (err && err.code === 'ETIMEDOUT') {
-          _this.errorHandler({
+          this.errorHandler({
```

Comment:
Since there are two types of errors (one where `err` is not null, and one where the `response.statusCode` is not between 200 and 300), if they are not handled differently they may as well be in the same conditional. ie. just have one `if/else` instead of two `if`s.